docs/CloudStorageToSnowflake-action.md

```diff
@@ -0,0 +1,183 @@
+# Cloud Storage to Snowflake Action
+
+
+Description
+-----------
+Loads data from external or internal staged files to an existing [Snowflake](https://www.snowflake.com/product/) table.
+
+Plugin works on top of COPY INTO. Please see its 
+[docs](https://docs.snowflake.com/en/sql-reference/sql/copy-into-table.html)
+for more information.
+
+Configuration
+-------------
+
+### Basic
+
+**Label:** Label for UI.
+
+**Account Name:** Snowflake account name and region and possibly a cloud specifier. (Part of the URL you use to 
+log in to Snowflake, minus the "snowflakecomputing.com"). E.g. "myaccount.us-central1.gcp".
+
+**Database:** Database that contains the target table.
+
+**Schema:** Schema that contains the target table.
+
+**Warehouse:** Warehouse to use for the data load operation (i.e. compute resources).
+
+**Role:** Role to use (e.g. `ACCOUNTADMIN`).
+
+**Source Type:** The source of the data to load. Possible values: `From Path`, `From Query`.
+
+**Source Path:** Internal or external location where the files containing data are staged.
+
+Examples of internal locations formats:
+```
+@~[/{path}]
+@[{namespace}.]{int_stage_name}[/{path}]
+@[{namespace}.]%{table_name}[/{path}]
+```
+
+Examples of external locations formats:
+```
+@[{namespace}.]{ext_stage_name}[/{path}]
+s3://{bucket}[/{path}]
+gcs://{bucket}[/{path}]
+azure://{account}.blob.core.windows.net/{container}[/{path}]
+```
+
+**Source Query:** An SQL query which specifies an explicit set of fields/columns (separated by commas) 
+to load from the staged data files. The fields/columns are selected from the files using a standard SQL query.
+ 
+Example:
+```
+SELECT name, address, phone from s3://bucket/path
+```
+
+**Destination Table:** Snowflake table to load data into.
+
+### Credentials
+
+**Username:** Username to use to connect to your Snowflake account.
+
+**Password:** Password to use to connect to your Snowflake account. Not necessary for key pair or OAuth2 authentication.
+
+### Key Pair Authentication
+
+**Key Pair Authentication Enabled:** If true, plugin will perform key pair authentication.
+
+**Private Key:** Private key contents.
+
+**Key File Passphrase:** Passphrase for the private key file.
```

Comment:
what if passphrase what not provided for key?

Reply:
We call this method from snowflake jdbc lib.
`  public void setPrivateKeyFile(String location, String password) {
    this.setAuthenticator("SNOWFLAKE_JWT");
    this.properties.put("private_key_file", location);
    if (!Strings.isNullOrEmpty(password)) {
      this.properties.put("private_key_file_pwd", password);
    }

  }`

The passphrase is not required in this case so if we pass null/empty string it is totally fine.